mono/mini/interp/transform.c

```diff
@@ -7241,30 +7264,29 @@ interp_cprop (TransformData *td)
 			ins = interp_fold_binop (td, sp, ins);
 			sp--;
 		} else if (ins->opcode >= MINT_STFLD_I1 && ins->opcode <= MINT_STFLD_P && (mono_interp_opt & INTERP_OPT_SUPER_INSTRUCTIONS)) {
-			if (sp [-2].ins) {
-				InterpInst *obj_ins = sp [-2].ins;
-				if (obj_ins->opcode == MINT_LDLOC_O) {
-					int loc_index = obj_ins->data [0];
+			StackContentInfo *src = &sp [-2];
+			if (src->ins) {
+				if (src->val.type == STACK_VALUE_LOCAL) {
+					int loc_index = src->val.local;
 					int fld_offset = ins->data [0];
 					int mt = ins->opcode - MINT_STFLD_I1;
 					ins = interp_insert_ins (td, ins, MINT_STLOCFLD_I1 + mt);
 					ins->data [0] = loc_index;
 					ins->data [1] = fld_offset;
+					local_ref_count [loc_index]++;
```

Comment:
@lewurm This is fishy. Assume you have a ldloc followed by a stfld. You would end up with the local_ref_count of this local at 2 after the end of the pass, even though it is used only once. I don't think it would have negative consequences but it is confusing for the code. I'd say the correct fix is to only increment it if the instruction that pushed that stack slot is not a LDLOC (which I guess would only happen with a dup ?, since only then we are inserting an additional access for the local)